include/pybind11/detail/function_record.h

```diff
@@ -477,7 +477,7 @@ struct args<Return, std::tuple<Args...>, std::tuple<Extra...>> {
                 }
                 // Separator for keyword-only arguments, placed before the kw
                 // arguments start (unless we are already putting an *args)
-                if (!has_args && arg_index == nargs_pos) {
+                if (PYBIND11_SILENCE_MSVC_C4127(!has_args && arg_index == nargs_pos)) {
```

Comment:
Only `has_args` should be wrapped.

Reply:
Will do. As an aside, can we just disable this warning period? It makes metaprogramming a pain ...

Reply:
It's probably also good to disable it to help the optimizer. We want the compiler to know these conditions are constant so they can removed from the resulting executable.